developer/js/source/lexical-model-compiler/lexical-model-compiler.ts

```diff
@@ -97,20 +92,36 @@ export class ModelSourceError extends Error {
  * Returns a JavaScript expression (as a string) that can serve as a word
  * breaking function.
  */
-function compileWordBreaker(wordBreakerSpec: WordBreakerSpec) {
-  // Use the default word breaker when it's unspecified
-  if (!wordBreakerSpec) {
-    wordBreakerSpec = 'default';
-  }
-
-  if (typeof wordBreakerSpec === "string") {
+function compileWordBreaker(spec: WordBreakerSpec): string {
+  if (typeof spec.use === "string") {
     // It must be a builtin word breaker, so just instantiate it.
-    return `wordBreakers['${wordBreakerSpec}']`;
+    return `wordBreakers['${spec.use}']`;
   } else {
-    // The word breaker was passed as a literal function; use its source code.
-    return wordBreakerSpec.toString()
+    return spec.use.toString()
       // Note: the .toString() might just be the property name, but we want a
       // plain function:
       .replace(/^wordBreak(ing|er)\b/, 'function');
   }
 }
+
+/**
+ * Given a word breaker specification in any of the messy ways,
+ * normalizes it to a common form that the compiler can deal with.
+ */
+function normalizeWordBreakerSpec(wordBreakerSpec: LexicalModelSource["wordBreaker"]): WordBreakerSpec {
+  if (wordBreakerSpec == undefined) {
```

Comment:
Previously you tested `!wordBreakerSpec`. Is that identical to `wordBreakerSpec == undefined`?

Reply:
`wordBreaker == undefined` is true when `wordBreaker === null` or `wordBreaker === undefined`, and is false otherwise. `!wordBreaker` is a bit more permissive in that it allows for `wordBreaker === 0` and `wordBreaker === ''`.

Reply:
I got bored so I decided to make a truth table:

`v`         | `v == undefined` | `!v`   | Desired
------------|------------------|--------|------------
`undefined` | `true`           | `true` | `true`
`null`      | `true`           | `true` | `true`
`0`         | `false`          | `true` | `false`
`false`     | `false`          | `true` | `false`
`''`        | `false`          | `true` | `false`
`NaN`       | `false`          | `true` | `false`

Basically, `v == undefined` gives me the behaviour I want: only `true` when wordBreaker is undefined or null.